controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -251,16 +248,16 @@ public Application createApplication(ApplicationId id, Optional<NToken> token) {
                 throw new IllegalArgumentException("Could not create '" + id + "': Application already exists");
             if (get(dashToUnderscore(id)).isPresent()) // VESPA-1945
                 throw new IllegalArgumentException("Could not create '" + id + "': Application " + dashToUnderscore(id) + " already exists");
-            if (tenant.get().isAthensTenant() && ! token.isPresent())
-                throw new IllegalArgumentException("Could not create '" + id + "': No NToken provided");
             if (tenant.get().isAthensTenant()) {
+                if ( ! token.isPresent())
+                throw new IllegalArgumentException("Could not create '" + id + "': No NToken provided");
+
                 ZmsClient zmsClient = zmsClientFactory.createZmsClientWithAuthorizedServiceToken(token.get());
                 try {
                     zmsClient.deleteApplication(tenant.get().getAthensDomain().get(),
```

Comment:
Please remove the deleteApplication call to Athenz. It's a workaround for an earlier Athenz issue that should be fixed now.

Reply:
Great!